libraries/common/src/main/java/androidx/media3/common/util/Util.java

```diff
@@ -1882,6 +1882,14 @@ public static int getAudioTrackChannelConfig(int channelCount) {
         return AudioFormat.CHANNEL_OUT_5POINT1 | AudioFormat.CHANNEL_OUT_BACK_CENTER;
       case 8:
         return AudioFormat.CHANNEL_OUT_7POINT1_SURROUND;
+      case 10:
+        if (Util.SDK_INT > 31) {
+          return AudioFormat.CHANNEL_OUT_5POINT1POINT4;
+        } else {
+          // This is used by DTS:X P2 with Direct Passthrough Playback.
+          // Specifying the audio format as 7.1 for 10 channels does not affect the playback.
+          return AudioFormat.CHANNEL_OUT_7POINT1_SURROUND;
+        }
```

Comment:
This change conflicts with an ongoing pull request from Dolby in which the case 10 will always return 5.1.4. I will consult with the platform team today what can happen (for SDK <= 31) when creating `AudioTrack` with channel count 5.1.4 for DTS:X P2 stream, or with channel count 7.1 for DD+JOC stream.

If there is any potential problem with that, I'm afraid that we should refactor this method `getAudioTrackChannelConfig` first to take the mime type into consideration.

Reply:
Creating an AudioTrack for DTS:X P2 with AudioFormat.CHANNEL_OUT_5POINT1POINT4 (taking the integer value) will fail on MTK platforms. That is why I switched it to AudioFormat.CHANNEL_OUT_7POINT1_SURROUND for API <=31. :)

Reply:
Acknowledged, we will discuss internally tomorrow on the potential refactorization of `getAudioTrackChannelConfig`. Thanks!

Reply:
@tianyif Any updates on this? Thanks.

Reply:
We got a hint from the audio framework team that the height channel masks were added in SDK 32 (e.g. [CHANNEL_OUT_TOP_BACK_LEFT](https://developer.android.com/reference/android/media/AudioFormat#CHANNEL_OUT_TOP_BACK_LEFT)), so we are now reaching to Dolby to see AudioTrack creation with 5.1.4 will also fail for them when API <= 31. If that's the case, the conflicts won't be there. But we are also waiting for a concrete answer from audio framework team as well.